src/Functions/URL/topLevelDomain.cpp

```diff
@@ -28,7 +28,7 @@ struct ExtractTopLevelDomain
                 return;
 
             /// For IPv4 addresses select nothing.
-            if (last_dot[1] <= '9')
+            if (last_dot[1] >= '0' && last_dot[1] <= '9')
```

Comment:
isNumericASCII

Reply:
Maybe also add
- a comment that it expects null terminated string?

Reply:
> a comment that it expects null terminated string?

Added another comment (about getURLHost())

>isNumericASCII

Right, done